torchao/experimental/ops/tests/test_linear_8bit_act_xbit_weight.cpp

```diff
@@ -36,14 +41,40 @@ UKernelConfig get_ukernel_config() {
       &ukernel::prepare_weight_data<weight_nbit, has_weight_zeros, has_bias>;
   config.kernel_fn =
       &ukernel::kernel<weight_nbit, has_weight_zeros, has_bias, has_clamp>;
+  } else {
+#if defined(TORCHAO_ENABLE_KLEIDI)
+    assert (weight_nbit == 4);
+    assert (!has_weight_zeros);
+
+    namespace kernel = torchao::kernels::cpu::aarch64::kleidi::
+      kai_matmul_clamp_f32_qai8dxp_qsi4c32p::neon_dotprod_1x8x32;
+
+    auto uk = kernel::get_ukernel();
+    config.mr = uk.get_mr();
+    config.nr = uk.get_nr();
+
+    config.activation_data_size_fn = &kernel::activation_data_size;
+    config.weight_data_size_fn = &kernel::weight_data_size;
+
+    config.preferred_activation_data_alignment = kernel::get_preferred_alignement();
+    config.preferred_weight_data_alignment = kernel::get_preferred_alignement();
+
+    config.prepare_activation_data_fn = &kernel::prepare_activation_data;
+    config.prepare_weight_data_fn = &kernel::prepare_weight_data; 
+
+    config.kernel_fn = &kernel::kernel;
+#else
+    assert (false);
+#endif // TORCHAO_ENABLE_KLEIDI
+  }
 
   return config;
 }
 
-template <int weight_nbit, bool has_weight_zeros, bool has_bias, bool has_clamp>
+template <int weight_nbit, bool has_weight_zeros, bool has_bias, bool has_clamp, bool has_kleidi=false>
```

Comment:
What do you think of changing void test_linear_8bit_act_xbit_weight(int m, int n, int k, int group_size) to void test_linear_8bit_act_xbit_weight(int m, int n, int k, int group_size, UkernelConfig config).

And then create 

template <int weight_nbit, bool has_weight_zeros, bool has_bias, bool has_clamp>
UKernelConfig get_ukernel_config_universal()

and 

UKernelConfig get_ukernel_config_kleidi()

and use these in the tests.

Reply:
yeah I like it, esp when we increase number of uKernel configs. Ok if we do it in the next PR?

Reply:
At least create a task for it in the lowbit kernel GSD so we don't forget

Reply:
[T206115417](https://www.internalfb.com/intern/tasks/?t=206115417)